src/cpp/src/speculative_decoding/speculative_decoding_eagle3_impl.cpp

```diff
@@ -0,0 +1,327 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+#include "speculative_decoding_eagle3_impl.hpp"
+#include "logger.hpp"
+
+namespace ov::genai {
+void share_embedding_weights(std::shared_ptr<ov::Model>& main_model, std::shared_ptr<ov::Model>& draft_model) {
```

Comment:
```suggestion
void share_embedding_weights(const std::shared_ptr<ov::Model>& main_model, std::shared_ptr<ov::Model>& draft_model) {
```

you share weights from the main model to the draft model. Please leave a comment what are these weights, and why we share it to draft model